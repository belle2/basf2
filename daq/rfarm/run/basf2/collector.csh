#! /bin/tcsh
basf2 collector.py $1 $2
exit

