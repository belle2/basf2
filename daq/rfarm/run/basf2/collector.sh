#! /bin/bash

# Set trap
trap "echo signal; killall basf2; exit" INT TERM HUP

# Run basf2
basf2 --no-stats collector.py $1 $2 &
#basf2 --no-stats roitest.py $1 $2 &

# For Debugging with valgrind
#valgrind --leak-check=full basf2 --no-stats collector.py $1 $2 &

# For Debugging using gdb, cannot stop
#gdb basf2 <<EOF
#run --no-stats collector.py $1 $2
#bt
#frame 0
#p (*(EvtHeader*)m_data)
#quit
#EOF

# Waiting for the completion forever
while :;do
  sleep 2
done

