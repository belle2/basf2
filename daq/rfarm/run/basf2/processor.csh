#! /bin/tcsh
onintr emergency
@ ncore=`cat /proc/cpuinfo | grep "processor" | wc | awk '{print $1}'` / 2
basf2 recon.py $1 $2 $3 $ncore
exit
emergency:
echo "interrupt received"
killall basf2
exit

