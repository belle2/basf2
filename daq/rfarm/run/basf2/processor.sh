#! /bin/bash
trap "echo signal; killall basf2; exit" INT TERM HUP
#ncore=`cat /proc/cpuinfo | grep "processor" | wc | awk '{print $1}'` / 2
ncore=8
basf2 --no-stats processor.py $1 $2 $3 $ncore &
#basf2 --no-stats roitest.py $1 $2 $3 $ncore &

while :;do
  sleep 2
done
