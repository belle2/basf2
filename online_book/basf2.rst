.. _onlinebook_basf2:

Working with Belle II software.
===============================

This chapter contains lessons designed to get you up-and-running using the
Belle II software to perform analysis tasks.

.. toctree::

   basf2/introduction
   basf2/first_steering_file
   basf2/roe
   basf2/various_additions
   basf2/flavor_tagging
   basf2/vertex_fitting
   basf2/event_display
   basf2/generating_mc
   basf2/fei
   basf2/cs
   basf2/b2bii
   basf2/skim
   basf2/python_module
