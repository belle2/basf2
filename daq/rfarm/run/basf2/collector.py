#!/usr/bin/env python
# -*- coding: utf-8 -*-

import os
import sys

from basf2 import *
from simulation import register_simulation
from reconstruction import register_reconstruction

set_log_level(LogLevel.ERROR)

argvs = sys.argv
argc = len(argvs)

print argvs[1]
print argc

# detecor components to be reconstructed
components = [
    'MagneticField',
    'BeamPipe',
    'PXD',
    'SVD',
    'CDC',
    'TOP',
    'ARICH',
    'BKLM',
    'ECL',
    ]

# Register modules to declare objects
register_simulation(components)
register_reconstruction(components)

# create a main path
main = create_path()

# Add input module
# input = register_module("SeqRootInput")
# input.param ( "inputFileName", "/fcdisk1-1/data/sim/sim-evtgen.sroot")
# main.add_module(input)

# Add output module
# output= register_module("SeqRootOutput")
# output.param ( "outputFileName", "/dev/null" )
# main.add_module(output)

# Add Rbuf2Ds
rbuf2ds = register_module('Rbuf2Ds')
rbuf2ds.param('RingBufferName', argvs[1])
main.add_module(rbuf2ds)

# Add Progress
progress = register_module('Progress')
main.add_module(progress)

# Add Ds2Rbuf
# ds2rbuf = register_module("Ds2Rbuf")
# ds2rbuf.param("RingBufferName", argvs[2])
# main.add_module(ds2rbuf)

# Run
process(main)

