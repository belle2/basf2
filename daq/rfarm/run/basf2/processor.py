#!/usr/bin/env python
# -*- coding: utf-8 -*-

import os
import sys

from basf2 import *

# command arguments
#   argvs[1] = input ring buffer name
#   argvs[2] = output ring buffer name
#   argvs[3] = port number of hserver
#   argvs[4] = number of cores for parallel processing
argvs = sys.argv
argc = len(argvs)

# set_log_level(LogLevel.INFO)
set_log_level(LogLevel.ERROR)

# path create
main = create_path()

# Rbuf2Ds as input module
# rbuf2ds = register_module("Rbuf2Ds")
# rbuf2ds.param("RingBufferName", argvs[1])
# main.add_module(rbuf2ds)

# Raw2Ds as input module
raw2ds = register_module('Raw2Ds')
raw2ds.param('RingBufferName', argvs[1])
main.add_module(raw2ds)

# Histo Module
# histo = register_module('HistoManager')
# main.add_module (histo)
histo = register_module('DqmHistoManager')
histo.param('Port', 9991)
histo.param('DumpInterval', 200)
main.add_module(histo)

# Monitor module
monitor = register_module('MonitorData')
main.add_module(monitor)

# Ds2Rbuf as output module
ds2rbuf = register_module('Ds2Rbuf')
ds2rbuf.param('RingBufferName', argvs[2])
main.add_module(ds2rbuf)

# Progress
progress = register_module('Progress')
main.add_module(progress)

# Debug modules
elapsed = register_module('ElapsedTime')
main.add_module(elapsed)

# Process events
set_nprocesses(int(argvs[4]))
process(main)
