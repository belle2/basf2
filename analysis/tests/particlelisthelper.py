#!/usr/bin/env python3

##########################################################################
# basf2 (Belle II Analysis Software Framework)                           #
# Author: The Belle II Collaboration                                     #
#                                                                        #
# See git log for contributors and copyright holders.                    #
# This file is licensed under LGPL-3.0, see LICENSE.md.                  #
##########################################################################

"""
Simple script to create some particles in a list with a python module
"""

# Go away doxygen, no one likes you
# @cond

import basf2
from ROOT import Belle2
from ROOT.Math import PxPyPzEVector


class MyParticleCreator(basf2.Module):
    def initialize(self):
        self.listB0 = Belle2.ParticleListHelper("B0:all")
        self.listPi0 = Belle2.ParticleListHelper("pi0:all")
        print(self.listB0.isSelfConjugated(), self.listPi0.isSelfConjugated())

    def event(self):
        self.listB0.create()
        self.listPi0.create()
        for i in range(5):
            self.listB0.addParticle(PxPyPzEVector(i, 0, 0, i), i % 2 == 0)
            self.listPi0.addParticle(PxPyPzEVector(i, 0, 0, i), i % 2 == 0)


basf2.set_random_seed("something important")
path = basf2.create_path()
path.add_module("EventInfoSetter", evtNumList=[2])
path.add_module(MyParticleCreator())
path.add_module("ParticlePrinter", listName="B0:all")
path.add_module("ParticlePrinter", listName="pi0:all")
basf2.process(path)

# @endcond
