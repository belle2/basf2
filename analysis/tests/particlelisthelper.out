[INFO] Steering file: analysis/tests/particlelisthelper.py
[INFO] The random number seed is set to "something important"
[INFO] Starting event processing, random seed is set to 'something important'
False True
[INFO] [ParticlePrinterModule] START ------------------------------
[INFO]  ParticleLists: B0:all (2+0) + anti-B0:all (3+0)
[INFO] collection=Particles
 PDGCode=511 Charge=0 PDGMass=5.28
 flavorType=1 particleSource=0 particleTypeUsedForFit=0
 mdstIndex=0 arrayIndex=2 identifier=-1 daughterIndices:  (none)
 mass=0
 momentum=(1.00, 0.00, 0.00) p=1
 momentum scaling factor=1
 Energy loss correction=0
 position=(0.00, 0.00, 0.00)
 p-value of fit (if done): -1
 error matrix (px, py, pz, E, x, y ,z):
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0

 extra info=( ) 

[INFO] collection=Particles
 PDGCode=511 Charge=0 PDGMass=5.28
 flavorType=1 particleSource=0 particleTypeUsedForFit=0
 mdstIndex=0 arrayIndex=6 identifier=-1 daughterIndices:  (none)
 mass=0
 momentum=(3.00, 0.00, 0.00) p=3
 momentum scaling factor=1
 Energy loss correction=0
 position=(0.00, 0.00, 0.00)
 p-value of fit (if done): -1
 error matrix (px, py, pz, E, x, y ,z):
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0

 extra info=( ) 

[INFO] collection=Particles
 PDGCode=-511 Charge=0 PDGMass=5.28
 flavorType=1 particleSource=0 particleTypeUsedForFit=0
 mdstIndex=0 arrayIndex=0 identifier=-1 daughterIndices:  (none)
 mass=0
 momentum=(0.00, 0.00, 0.00) p=0
 momentum scaling factor=1
 Energy loss correction=0
 position=(0.00, 0.00, 0.00)
 p-value of fit (if done): -1
 error matrix (px, py, pz, E, x, y ,z):
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0

 extra info=( ) 

[INFO] collection=Particles
 PDGCode=-511 Charge=0 PDGMass=5.28
 flavorType=1 particleSource=0 particleTypeUsedForFit=0
 mdstIndex=0 arrayIndex=4 identifier=-1 daughterIndices:  (none)
 mass=0
 momentum=(2.00, 0.00, 0.00) p=2
 momentum scaling factor=1
 Energy loss correction=0
 position=(0.00, 0.00, 0.00)
 p-value of fit (if done): -1
 error matrix (px, py, pz, E, x, y ,z):
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0

 extra info=( ) 

[INFO] collection=Particles
 PDGCode=-511 Charge=0 PDGMass=5.28
 flavorType=1 particleSource=0 particleTypeUsedForFit=0
 mdstIndex=0 arrayIndex=8 identifier=-1 daughterIndices:  (none)
 mass=0
 momentum=(4.00, 0.00, 0.00) p=4
 momentum scaling factor=1
 Energy loss correction=0
 position=(0.00, 0.00, 0.00)
 p-value of fit (if done): -1
 error matrix (px, py, pz, E, x, y ,z):
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0

 extra info=( ) 

[INFO] [ParticlePrinterModule] END   ------------------------------
[INFO] [ParticlePrinterModule] START ------------------------------
[INFO]  ParticleList : pi0:all (0+5)
[INFO] collection=Particles
 PDGCode=111 Charge=0 PDGMass=0.135
 flavorType=0 particleSource=0 particleTypeUsedForFit=0
 mdstIndex=0 arrayIndex=1 identifier=-1 daughterIndices:  (none)
 mass=0
 momentum=(0.00, 0.00, 0.00) p=0
 momentum scaling factor=1
 Energy loss correction=0
 position=(0.00, 0.00, 0.00)
 p-value of fit (if done): -1
 error matrix (px, py, pz, E, x, y ,z):
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0

 extra info=( ) 

[INFO] collection=Particles
 PDGCode=111 Charge=0 PDGMass=0.135
 flavorType=0 particleSource=0 particleTypeUsedForFit=0
 mdstIndex=0 arrayIndex=3 identifier=-1 daughterIndices:  (none)
 mass=0
 momentum=(1.00, 0.00, 0.00) p=1
 momentum scaling factor=1
 Energy loss correction=0
 position=(0.00, 0.00, 0.00)
 p-value of fit (if done): -1
 error matrix (px, py, pz, E, x, y ,z):
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0

 extra info=( ) 

[INFO] collection=Particles
 PDGCode=111 Charge=0 PDGMass=0.135
 flavorType=0 particleSource=0 particleTypeUsedForFit=0
 mdstIndex=0 arrayIndex=5 identifier=-1 daughterIndices:  (none)
 mass=0
 momentum=(2.00, 0.00, 0.00) p=2
 momentum scaling factor=1
 Energy loss correction=0
 position=(0.00, 0.00, 0.00)
 p-value of fit (if done): -1
 error matrix (px, py, pz, E, x, y ,z):
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0

 extra info=( ) 

[INFO] collection=Particles
 PDGCode=111 Charge=0 PDGMass=0.135
 flavorType=0 particleSource=0 particleTypeUsedForFit=0
 mdstIndex=0 arrayIndex=7 identifier=-1 daughterIndices:  (none)
 mass=0
 momentum=(3.00, 0.00, 0.00) p=3
 momentum scaling factor=1
 Energy loss correction=0
 position=(0.00, 0.00, 0.00)
 p-value of fit (if done): -1
 error matrix (px, py, pz, E, x, y ,z):
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0

 extra info=( ) 

[INFO] collection=Particles
 PDGCode=111 Charge=0 PDGMass=0.135
 flavorType=0 particleSource=0 particleTypeUsedForFit=0
 mdstIndex=0 arrayIndex=9 identifier=-1 daughterIndices:  (none)
 mass=0
 momentum=(4.00, 0.00, 0.00) p=4
 momentum scaling factor=1
 Energy loss correction=0
 position=(0.00, 0.00, 0.00)
 p-value of fit (if done): -1
 error matrix (px, py, pz, E, x, y ,z):
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0

 extra info=( ) 

[INFO] [ParticlePrinterModule] END   ------------------------------
[INFO] [ParticlePrinterModule] START ------------------------------
[INFO]  ParticleLists: B0:all (2+0) + anti-B0:all (3+0)
[INFO] collection=Particles
 PDGCode=511 Charge=0 PDGMass=5.28
 flavorType=1 particleSource=0 particleTypeUsedForFit=0
 mdstIndex=0 arrayIndex=2 identifier=-1 daughterIndices:  (none)
 mass=0
 momentum=(1.00, 0.00, 0.00) p=1
 momentum scaling factor=1
 Energy loss correction=0
 position=(0.00, 0.00, 0.00)
 p-value of fit (if done): -1
 error matrix (px, py, pz, E, x, y ,z):
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0

 extra info=( ) 

[INFO] collection=Particles
 PDGCode=511 Charge=0 PDGMass=5.28
 flavorType=1 particleSource=0 particleTypeUsedForFit=0
 mdstIndex=0 arrayIndex=6 identifier=-1 daughterIndices:  (none)
 mass=0
 momentum=(3.00, 0.00, 0.00) p=3
 momentum scaling factor=1
 Energy loss correction=0
 position=(0.00, 0.00, 0.00)
 p-value of fit (if done): -1
 error matrix (px, py, pz, E, x, y ,z):
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0

 extra info=( ) 

[INFO] collection=Particles
 PDGCode=-511 Charge=0 PDGMass=5.28
 flavorType=1 particleSource=0 particleTypeUsedForFit=0
 mdstIndex=0 arrayIndex=0 identifier=-1 daughterIndices:  (none)
 mass=0
 momentum=(0.00, 0.00, 0.00) p=0
 momentum scaling factor=1
 Energy loss correction=0
 position=(0.00, 0.00, 0.00)
 p-value of fit (if done): -1
 error matrix (px, py, pz, E, x, y ,z):
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0

 extra info=( ) 

[INFO] collection=Particles
 PDGCode=-511 Charge=0 PDGMass=5.28
 flavorType=1 particleSource=0 particleTypeUsedForFit=0
 mdstIndex=0 arrayIndex=4 identifier=-1 daughterIndices:  (none)
 mass=0
 momentum=(2.00, 0.00, 0.00) p=2
 momentum scaling factor=1
 Energy loss correction=0
 position=(0.00, 0.00, 0.00)
 p-value of fit (if done): -1
 error matrix (px, py, pz, E, x, y ,z):
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0

 extra info=( ) 

[INFO] collection=Particles
 PDGCode=-511 Charge=0 PDGMass=5.28
 flavorType=1 particleSource=0 particleTypeUsedForFit=0
 mdstIndex=0 arrayIndex=8 identifier=-1 daughterIndices:  (none)
 mass=0
 momentum=(4.00, 0.00, 0.00) p=4
 momentum scaling factor=1
 Energy loss correction=0
 position=(0.00, 0.00, 0.00)
 p-value of fit (if done): -1
 error matrix (px, py, pz, E, x, y ,z):
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0

 extra info=( ) 

[INFO] [ParticlePrinterModule] END   ------------------------------
[INFO] [ParticlePrinterModule] START ------------------------------
[INFO]  ParticleList : pi0:all (0+5)
[INFO] collection=Particles
 PDGCode=111 Charge=0 PDGMass=0.135
 flavorType=0 particleSource=0 particleTypeUsedForFit=0
 mdstIndex=0 arrayIndex=1 identifier=-1 daughterIndices:  (none)
 mass=0
 momentum=(0.00, 0.00, 0.00) p=0
 momentum scaling factor=1
 Energy loss correction=0
 position=(0.00, 0.00, 0.00)
 p-value of fit (if done): -1
 error matrix (px, py, pz, E, x, y ,z):
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0

 extra info=( ) 

[INFO] collection=Particles
 PDGCode=111 Charge=0 PDGMass=0.135
 flavorType=0 particleSource=0 particleTypeUsedForFit=0
 mdstIndex=0 arrayIndex=3 identifier=-1 daughterIndices:  (none)
 mass=0
 momentum=(1.00, 0.00, 0.00) p=1
 momentum scaling factor=1
 Energy loss correction=0
 position=(0.00, 0.00, 0.00)
 p-value of fit (if done): -1
 error matrix (px, py, pz, E, x, y ,z):
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0

 extra info=( ) 

[INFO] collection=Particles
 PDGCode=111 Charge=0 PDGMass=0.135
 flavorType=0 particleSource=0 particleTypeUsedForFit=0
 mdstIndex=0 arrayIndex=5 identifier=-1 daughterIndices:  (none)
 mass=0
 momentum=(2.00, 0.00, 0.00) p=2
 momentum scaling factor=1
 Energy loss correction=0
 position=(0.00, 0.00, 0.00)
 p-value of fit (if done): -1
 error matrix (px, py, pz, E, x, y ,z):
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0

 extra info=( ) 

[INFO] collection=Particles
 PDGCode=111 Charge=0 PDGMass=0.135
 flavorType=0 particleSource=0 particleTypeUsedForFit=0
 mdstIndex=0 arrayIndex=7 identifier=-1 daughterIndices:  (none)
 mass=0
 momentum=(3.00, 0.00, 0.00) p=3
 momentum scaling factor=1
 Energy loss correction=0
 position=(0.00, 0.00, 0.00)
 p-value of fit (if done): -1
 error matrix (px, py, pz, E, x, y ,z):
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0

 extra info=( ) 

[INFO] collection=Particles
 PDGCode=111 Charge=0 PDGMass=0.135
 flavorType=0 particleSource=0 particleTypeUsedForFit=0
 mdstIndex=0 arrayIndex=9 identifier=-1 daughterIndices:  (none)
 mass=0
 momentum=(4.00, 0.00, 0.00) p=4
 momentum scaling factor=1
 Energy loss correction=0
 position=(0.00, 0.00, 0.00)
 p-value of fit (if done): -1
 error matrix (px, py, pz, E, x, y ,z):
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0
          0          0          0          0          0          0          0

 extra info=( ) 

[INFO] [ParticlePrinterModule] END   ------------------------------
